src/components/TokenExpiredTracker.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { SessionTimeout } from 'patternfly-react'
+
+import AppConfiguration from '_/config'
+import ErrorAlert from './ErrorAlert'
+import { msg } from '_/intl'
+
+import { logout } from '_/actions'
+
+const TIME_TO_DISPLAY_MODAL = 30 // 30 seconds
+
+// TODO: allow the user to cancel the automatic reload?
+// If so, change config.isTokenExpired to false and add additional check to doCheckTokenExpired() before actual reload
+class TokenExpiredTracker extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expired: false,
+      counter: AppConfiguration.maxUserInactiveTime,
+    }
+
+    this.dropCounter = this.dropCounter.bind(this)
+    this.decrementCounter = this.decrementCounter.bind(this)
+
+    document.body.addEventListener('mousemove', this.dropCounter)
+    this.timer = setInterval(this.decrementCounter, 1000)
+  }
+
+  dropCounter () {
+    if (!this.state.expired) {
+      this.setState({ counter: AppConfiguration.maxUserInactiveTime })
+    }
+  }
+  decrementCounter () {
+    const state = {
+      counter: this.state.counter - 1,
+    }
+    if (this.state.counter <= TIME_TO_DISPLAY_MODAL && !this.state.expired) {
+      state.expired = true
+    }
+    if (this.state.counter <= 0) {
+      this.props.onLogout()
+    }
+    this.setState(state)
+  }
+
+  componentWillUnmount () {
+    clearInterval(this.timer)
+  }
+
+  render () {
+    const { config, onLogout } = this.props
+    if (this.state.expired) {
+      return <SessionTimeout
+        timeLeft={this.state.counter}
+        displayBefore={TIME_TO_DISPLAY_MODAL}
+        continueFnc={() => this.setState({ expired: false, counter: AppConfiguration.maxUserInactiveTime })}
+        logoutFnc={onLogout}
+        primaryContent={<p className='lead'>{ msg.sessionExpired() }</p>}
+        secondaryContent={
+          <React.Fragment>
+            <p>{ msg.logOutIn30SecondsSecondary() }</p>
```

Comment:
Thoughts on ticking down the seconds on the modal instead of a static "you have 30 seconds (or less)" message?